source/fuzz/transformation_access_chain.cpp

```diff
@@ -225,16 +251,55 @@ void TransformationAccessChain::Apply(
       // It is not a struct: the index will need clamping.
 
       // Get two new ids to use and update the amount used.
-      protobufs::UInt32Pair pair =
+      protobufs::UInt32Pair fresh_ids =
           message_.fresh_ids_for_clamping()[id_pairs_used++];
-      std::pair<uint32_t, uint32_t> fresh_ids;
-      fresh_ids.first = pair.first();
-      fresh_ids.second = pair.second();
 
-      new_index_id =
-          CreateAndGetClampedIndexId(ir_context, index_id, subobject_type_id,
-                                     true, fresh_ids)
-              .second;
+      // Perform the clamping using the fresh ids at our disposal.
+      // The module will not be changed if |add_clamping_instructions| is not
+      // set.
+      auto index_instruction = ir_context->get_def_use_mgr()->GetDef(index_id);
+
+      uint32_t bound = fuzzerutil::GetBoundForCompositeIndex(
+          *ir_context->get_def_use_mgr()->GetDef(subobject_type_id),
+          ir_context);
+
+      auto bound_minus_one_id =
+          fuzzerutil::MaybeGetIntegerConstantFromValueAndType(
+              ir_context, bound - 1, index_instruction->type_id());
+
+      opt::analysis::Bool bool_type;
+      uint32_t bool_type_id = ir_context->get_type_mgr()->GetId(&bool_type);
```

Comment:
Use MaybeGetBoolType, and add a precondition assertion.

Reply:
Done